os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c

```diff
@@ -116,7 +116,7 @@ void pg_timer_int_handler(void *Data)
 {
 	pmvdbg("PM Timer interrupt handler!!\n");
 	// Switch status back to normal mode after wake up from interrupt
-	pm_activity(PM_IDLE_DOMAIN, 9);	
+	pm_changestate(pm_checkstate());
```

Comment:
Do we need to call pm_checkstate in here?
can't we call pm_chagestate(PM_STANDBY)?

Reply:
Suppose if we are in PM_NORMAL state and some interrupt will come. If we do pm_changestate(PM_STANDBY), then our  LCD will switch OFF directly without going to LCD_DIM stage.

Reply:
I think changing state when back from sleep (ie. either to standby and normal), can be moved to idle thread logic in amebasmart_idle.c. As to previous implementation, different peripheral has change state as to their callback handler or it's own wakeup logic, and it seems not convenient for maintaining.
Thus, I think it might be better if we leave that to be done in the idle thread.